### PR TITLE
adding global allowEmpty config 

### PR DESCRIPTION
**Pull request by jcope2013**
as discussed in https://github.com/stefanpenner/ember-moment/issues/84, an allowEmpty global config to be used throughout application, let me know your thoughts

### CODE REVIEW

**Comment by jasonmit** [comment]
Tests are failing on beta and canary but same thing happens against master so will merge.  Will create another issue to resolve this.

**Comment by jasonmit** [comment]
Can't wait to drop support for < 1.13.. we should do this soon to clean up all this factory method madness

**Comment by jcope2013** [comment]
@jasonmit yeah i was wondering your thoughts, can ember-try programmatically remove the initializer on certain versions?

```

// This initializer is only included in applications that are <= Ember 1.12.0
// Otherwise, it is programmatically removed during the build process via `index.js` `treeForApp` method

https://github.com/stefanpenner/ember-moment/blob/master/app/initializers/legacy-register.js
```

**Comment by jasonmit** [comment]
> ember-try programmatically remove the initializer on certain versions?

Not that I am aware of, we are the ones moving/renaming the initializer in `index.js`

File: addon/helpers/deprecated/ago.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import Ember from 'ember';
-import { helperFactory } from '../moment-from-now';
+import momentFromNowHelper from '../moment-from-now';
```
Comment (jasonmit) [inline addon/helpers/deprecated/ago.js]: This breaks 1.10.0.  Causes the Ember.deprecate to immediately fire even when not using {{ago}}

**Comment by jasonmit** [comment]
I'll fix the Ember.deprecate issue by reverting your removal of helperFactory and put a comment there explaining why it's needed.

**Pull request merged**